assembl/static2/js/app/components/cookies/cookieSetter.jsx

```diff
@@ -0,0 +1,72 @@
+// @flow
+/* global globalAnalytics */
+import React from 'react';
+import { I18n } from 'react-redux-i18n';
+import classnames from 'classnames';
+
+// Components
+import Helper from '../common/helper';
+import CookieSwitch from './cookieSwitch';
+
+import { COOKIE_TRANSLATION_KEYS } from '../../constants';
+
+export type CookieObject = {
+  name: string,
+  category: string,
+  accepted: boolean,
+  cookieType: string
+};
+
+type cookieSetterProps = {
+  handleToggle: Function,
+  cookie: CookieObject,
+  toggleCookieType: Function,
+  locale: string
+};
+
+// @$FlowFixMe globalAnalytics is a global variable
+const matomoHost = globalAnalytics.piwik ? globalAnalytics.piwik.host : null;
+
+const cookieSetter = ({ handleToggle, cookie, toggleCookieType, locale }: cookieSetterProps) => {
+  const toggleSwitch = () => {
+    const { accepted, cookieType } = cookie;
+    const updatedCookie = { ...cookie, accepted: !accepted, cookieType: toggleCookieType(cookieType) };
+    handleToggle(updatedCookie);
+  };
+
+  const { name, category, accepted } = cookie;
+
+  const cookieName = Object.keys(COOKIE_TRANSLATION_KEYS).includes(name) ? I18n.t(`cookies.${name}`) : name;
+  const cookieIsMatomo = name === 'matomo';
```

Comment:
Is it worth creating a constant in constants.js for 'matomo'?

Reply:
yeah, I agree, if it's a cookie, it should go to constants.
These two lines are a bit strange since you are declaring a `cookieName` const but then use `name` line 40. It looks like an error. If it's not you should move `cookieName` declaration to avoid this confusion

Reply:
oh ok, it seems that `cookieName` may be renamed `cookieLabel` or something like that